orttraining/orttraining/python/training/ortmodule/_training_manager.py

```diff
@@ -32,14 +32,17 @@ def execution_session_run_forward(execution_session, onnx_model, device, *inputs
         # Assert that the input and model device match
         _utils._check_same_device(device, "Input argument to forward", *inputs)
 
+        # Make sure inputs are contiguous, as a requirement converting it to ortvalue.
+        contiguous_inputs = _utils._to_contiguous(inputs)
```

Comment:
This can have a considerable performance hit. Pytorch delegates that call to the users as opposed to automatically calling it for perf reasons too.

Depending on the operations the model will execute, it is ok not to be contiguous. We should call it on a case by case basis

Currently we prefer to change the original training script to call contiguous API for specific tensors as opposed to pay the price  of calling it for all of them

Reply:
[peng]

Reply:
In understand the point "Depending on the operations the model will execute, it is ok not to be contiguous. We should call it on a case by case basis".

While in our case, here, you would see this input will be used by to_dlpack, which requires the inputs to be contigous.If we don't do this to_contigoush(), we need change the training script input as well. 

So I would suggest to add this logic, then we will have a bit less code change in user script when intergrating models.

Reply:
Maybe @kit1980 @mrry @SherlockNoMad can have numbers of perf hit by forcing contiguous() call to all tensors. My understading is that for some models ORTModule would become slower than pytorch

Reply:
Firstly, this is the inputs for forward runs. they are passed to ORT backend with to_dlpack. So if the inputs are not contiguous, this will fail with exception., so having it or not is a choice of let user model run or fail. If you check the backward inputs, we do the similar things. 

Secondly, whether we put the .contigous() here or in user training scripts, there is no big difference on perf, isn't it? 

If you feel we need more discussion needed, I would prefer to remove them from this PR to unblock the other changes. Instead, I would change the Megatron-LM model to do the .contiguous().

Reply:
revert the change to avoid more cycles to get other pieces in.

Reply:
> Firstly, this is the inputs for forward runs. they are passed to ORT backend with to_dlpack. So if the inputs are not contiguous, this will fail with exception., so having it or not is a choice of let user model run or fail. If you check the backward inputs, we do the similar things.
> 
> Secondly, whether we put the .contigous() here or in user training scripts, there is no big difference on perf, isn't it?
> 
My concern is that by always calling `contiguous()` we will incur in additional memory copy for models that do work without it, which affects performance. For models that would fail calling contiguous(), you are right. My point is: should we make all models pay the extra contiguous() call price to support the models that dont? @mrry  and @kit1980 started a thread discussing approaches to workaround this in the short term and some ideas for a long term fix. I dont have a strong preference, just trying to make sure we are being intentional in the changes with possible perf hit

> If you feel we need more discussion needed, I would prefer to remove them from this PR to unblock the other changes. Instead, I would change the Megatron-LM model to do the .contiguous().

@mrry any preference on whether keep the contiguous call for all tensors on ORTModule?